File: frontend/src/typings/GameCollection.ts
import ModelBase from 'App/ModelBase';
import Game, { Image, GameAvailability } from 'Game/Game';

interface GameCollection extends ModelBase {
  igdbId: number;
  sortTitle: string;
  title: string;
  overview: string;
  monitored: boolean;
  minimumAvailability: GameAvailability;
  qualityProfileId: number;
  rootFolderPath: string;
  searchOnAdd: boolean;
  images: Image[];
  games: Game[];
  missingGames: number;
  tags: number[];
  isSaving?: boolean;
}

export default GameCollection;

File: src/NzbDrone.Core/Games/Collections/GameCollection.cs
using System;
using System.Collections.Generic;
using NzbDrone.Core.Datastore;

namespace NzbDrone.Core.Games.Collections
{
    public class GameCollection : ModelBase
    {
        public GameCollection()
        {
            Images = new List<MediaCover.MediaCover>();
            Tags = new HashSet<int>();
        }

        public string Title { get; set; }
        public string CleanTitle { get; set; }
        public string SortTitle { get; set; }
        public int IgdbId { get; set; }
        public string Overview { get; set; }
        public bool Monitored { get; set; }
        public int QualityProfileId { get; set; }
        public string RootFolderPath { get; set; }
        public bool SearchOnAdd { get; set; }
        public GameStatusType MinimumAvailability { get; set; }
        public DateTime? LastInfoSync { get; set; }
        public List<MediaCover.MediaCover> Images { get; set; }
        public DateTime Added { get; set; }
        public List<GameMetadata> Games { get; set; }
        public HashSet<int> Tags { get; set; }

        public void ApplyChanges(GameCollection otherCollection)
        {
            IgdbId = otherCollection.IgdbId;

            Monitored = otherCollection.Monitored;
            SearchOnAdd = otherCollection.SearchOnAdd;
            QualityProfileId = otherCollection.QualityProfileId;
            MinimumAvailability = otherCollection.MinimumAvailability;
            RootFolderPath = otherCollection.RootFolderPath;
            Tags = otherCollection.Tags;
        }
    }
}
